vllm/worker/hpu_model_runner.py

```diff
@@ -255,6 +271,7 @@ def _regional_compilation(self,
                                            children_name)
 
     def _compile_region(self, model, name, module):
+        #print(f'compiling {module}...')
         module = torch.compile(module, backend='hpu_backend', dynamic=False)
```

Comment:
in order to use mark_dynamic. You need to enable dynamic shapes in torch.compile e.g. best if you don't pass dynamic=False flag.
mark_dynamic has some additional limitation. It is best to use it when you compile the entire model (without regional compilation)